mods/stairs/init.lua

```diff
@@ -32,7 +32,11 @@ local function rotate_and_place(itemstack, placer, pointed_thing)
 	if placer then
 		local placer_pos = placer:get_pos()
 		if placer_pos then
-			param2 = minetest.dir_to_facedir(vector.subtract(p1, placer_pos))
+			local vec = vector.subtract(p1, placer_pos)
+			param2 = minetest.dir_to_facedir(vec)
+			if math.abs(vec.z) <= 0.5 and  math.abs(vec.x) <= 0.5 and vec.y < 0 and p0.y == p1.y then
```

Comment:
This checks that:

* `math.abs(vec.x) <= 0.5 and math.abs(vec.z) <= 0.5`: Player is placing the node where he is standing? (XZ rectangle collision)
* `vec.y < 0`: Player is looking down? (feet are above placed node?)
* `p0.y == p1.y`: Stair was not placed above or under a node

Please remove the space before `math.abs` *and* name your variables properly *and* add some comments.

Reply:
Yes, all these statements are true.